src/engine/strat_engine/filesystem.rs

```diff
@@ -56,6 +59,52 @@ impl StratFilesystem {
         }
     }
 
+    /// Create a snapshot of the ThinDev under the filesystem. Return the
+    /// resulting ThinDev to the caller.
+    pub fn snapshot(&self,
+                    dm: &DM,
+                    thin_pool: &ThinPoolDev,
+                    snapshot_name: &DmName,
+                    snapshot_fs_uuid: FilesystemUuid,
+                    snapshot_thin_id: ThinDevId)
+                    -> EngineResult<ThinDev> {
+
+        match self.thin_dev
+                  .snapshot(dm, thin_pool, snapshot_name, snapshot_thin_id) {
+            Ok(thin_dev) => {
+                // If the source is mounted, XFS puts a dummy record in the
+                // log to enforce replay of the snapshot to deal with any
+                // orphaned inodes. The dummy record put the log in a dirty
+                // state. xfs_admin won't allow a filesystem UUID
+                // to be updated when the log is dirty.  To clear the log
+                // we mount/unmount the filesystem before updating the UUID.
+                //
+                // If the source is unmounted the XFS log will be clean so
+                // we can skip the mount/unmount.
+                if let Some(_origin_mount_point) = self.get_mount_point()? {
+                    let tmp_dir = TempDir::new("stratis_mp_")?;
```

Comment:
Can you just get the path here, instead of getting the path twice for the mount and umount calls,  as:

```let tmp_dir = TempDir::new("...").path(); ```

Reply:
It looks like if I do that the result of the TempDir::new() doesn't live long enough:

85 |                     let tmp_dir_path = TempDir::new("stratis_mp_")?.path();
   |                                        ----------------------------       ^ temporary value dropped here while still borrowed
   |                                        |
   |                                        temporary value created here

Reply:
Yes. Might as well leave it as is.